addons/payment/models/payment_token.py

```diff
@@ -108,3 +110,24 @@ def get_linked_records_info(self):
         """
         self.ensure_one()
         return []
+
+    def _get_token_display_name(self, *args, length=8, **kwargs):
+        """ Return a token name of the desired length and format. The default format is •••• 1111.
+        For custom display names of tokens for an acquirer, this method must be overridden.
+
+        Note: self.ensure_one()
+        Note: assert length <= 34
+
+        :param list args: Needed by qWeb when calling this method
+        :param int length: The desired final length of the token name (8 for a bank card)
+        :param dict kwargs: To add specific parameters to overrides of this method
+        :return: The padded token name
+        :rtype: str
+        """
+        self.ensure_one()
+        assert length <= 34  # The longest possible IBAN has 34 characters
+
+        if self.payment_details:
+            return f"{'•' *(length - 4)} {self.payment_details[-4:]}"
+        else:
+            return f"Payment details saved on {date.today().strftime('%Y/%m/%d')}"
```

Comment:
`today()` will make the customer believe that the token was saved... today. Use `self.create_date` instead.